assets/test/advanced_authoring/right_menu/screen/screen_mocks.ts

```diff
@@ -0,0 +1,64 @@
+import { IActivity } from "apps/delivery/store/features/activities/slice";
+
+export const transformedSchema = {
+  title: 'Customize Avatar',
+  width: 1000,
+  height: 500,
+  customCssClass: '',
+  combineFeedback: false,
+  showCheckBtn: false,
+  checkButtonLabel: '',
+  maxAttempt: 0,
+  maxScore: 0,
+  palette: {},
+  trapStateScoreScheme: true,
+  negativeScoreAllowed: false,
+  screenButton: false,
+};
+
+export const screen: IActivity = {
+  id: 29,
+  resourceId: 29,
+  activitySlug: 'customize_avatar_5we6z',
+  content: {
+    custom: {
+      applyBtnFlag: false,
+      applyBtnLabel: '',
+      combineFeedback: false,
+      facts: [],
+      lockCanvasSize: false,
+      mainBtnLabel: '',
+      negativeScoreAllowed: false,
+      palette: {
+        backgroundColor: "rgba(255,255,255,1)",
+        borderColor: "rgba(255,255,255,1)",
+        borderStyle: "solid",
+        borderWidth: "0.1px",
+        fillAlpha: 1,
+        fillColor: 16777215,
+        lineAlpha: 1,
+        lineColor: 16777215,
+        lineStyle: 0,
+        lineThickness: 0.1,
+        borderRadius: "10px",
+        useHtmlProps: false,
```

Comment:
think about the logic that is happening, and create scenarios to test it. so with the palette for example, what we want to happen is: if it encounters a falsey `useHtmlProps` then we want to convert it using the "old" properties like `fillColor` etc into the `backgroundColor`. If `useHtmlProps` is truthy then we want to simply use (and expect that they are there) the `backgroundColor` etc properties and ignore the others. We want to permanently convert them every time. So when it comes out of the editor after they make *any* change then it should have `useHtmlProps` set to true, and the `backgroundColor` and related properties should be filled out. We want to save that, so that next time they come in, it will not do the conversion anymore.

So we need to create multiple versions of screen data here, some that have `useHtmlProps` as falsey and some as truthy. The tests should make sure that the functions behave in line with the business rules above.

not only that, but the `transformScreenModeltoSchema` method transforms more than just the palette, so it should check that passing in a screen data custom object returns a properly formed schema object.

Reply:
I have implemented the palette related explanation you have mentioned. and other properties are covered with the method i have used i.e. toMatchObject(), which checks for the properties and its values.